incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/MqttKafkaConfiguration.java

```diff
@@ -105,106 +110,105 @@ public int willStreamReconnectDelay()
         return WILL_STREAM_RECONNECT_DELAY.getAsInt(this);
     }
 
-
-    private static String decodeStringSupplier(
+    private static StringSupplier decodeStringSupplier(
         Configuration config,
         String fullyQualifiedMethodName)
     {
-        Supplier<String> supplier = null;
+        StringSupplier supplier = null;
 
         try
         {
-            MethodType signature = MethodType.methodType(String.class);
+            MethodType signature = MethodType.methodType(StringSupplier.class);
```

Comment:
The referenced method is a `StringSupplier`, so it should return `String`, not `StringSupplier`.

Then the lambda returns `value` (String), not `value.get()`.

Same feedback for `decodeLongSupplier`.